gpu/README.md

```diff
@@ -0,0 +1,193 @@
+# Agent Check: GPU
+
+## Overview
+
+This check monitors GPU devices and their utilization through the Datadog Agent.
+
+Supported vendors: NVIDIA.
+
+- Track utilization of GPU devices and retrieve performance and health metrics.
+- Monitor processes that are using GPU devices and their performance.
+
+## Setup
+
+### Installation
+
+The GPU check is included in the [Datadog Agent][2] package. No additional installation is needed on your server.
+
+The check also uses eBPF probes to assign GPU usage and performance metrics to processes. eBPF programs are loaded by the `system-probe` component.
+
+**Note**: The `system-probe` GPU component (which generates per-process metrics) requires Linux kernel 5.8 or later. Windows is not supported.
+
+### Configuration
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "Host" xxx -->
+
+#### Host
+
+Enabling the `gpu` integration requires `system-probe` to have the configuration option enabled.  Inside the `system-probe.yaml` configuration file, the following parameters must be set:
+
+```yaml
+gpu_monitoring:
+  enabled: true
+```
+
+The check in the agent configuration file is enabled by default whenever NVIDIA GPUs and their drivers are detected in the system. However, it can also be configured manually following these steps:
+
+1. Edit the `gpu.d/conf.yaml` file, in the `conf.d/` folder at the root of your
+   Agent's configuration directory to start collecting your GPU performance data.
+   See the [sample gpu.d/conf.yaml][3] for all available configuration options.
+
+2. [Restart the Agent][4].
+
+This check is automatically enabled when the Agent is running on a host with NVIDIA GPUs and the NVIDIA drivers and libraries installed.
+
+<!-- xxz tab xxx -->
+<!-- xxx tab "Containerized" xxx -->
+
+#### Important: Running on Helm/Kubernetes in mixed environments
+
+One important thing to note in the deployment for Kubernetes clusters is that, in order to access the GPUs, the Datadog Agent pods will need access to both the GPUs and NVIDIA's NVML library (`libnvidia-ml.so`). Due to the design of NVIDIA's k8s device plugin, in order to have access to those features the agent pods will need to run with the `nvidia` runtime class. This means that the agent pods will not be able to run in the default runtime class.
+
+This can cause issues in clusters where some nodes have GPUs and others don't: if we deploy with a single runtime class the agent will only run on a subset of the cluster nodes. Both the Helm and Datadog Operator deployments can be configured to deploy in this situation correctly to both types of nodes, but it requires some additional configuration as described below.
```

Comment:
```suggestion
This can cause issues in clusters where some nodes have GPUs and others don't: if we deploy with a single runtime class, the Agent will only run on a subset of the cluster nodes. Both the Helm and Datadog Operator deployments can be configured to deploy in this situation correctly to both types of nodes, but it requires some additional configuration as described below.
```